src/hiopbbpy/opt/boalgorithm.py

```diff
@@ -175,31 +195,39 @@ def optimize(self):
           print(f"*****************************")
           print(f"Iteration {i+1}/{self.bo_maxiter}")
 
-          # Get a new sample point
-          x_new = self._find_best_point(x_train, y_train)
+          y_train_virtual = y_train.copy() # old training + batch_size num of virtual points
+          for batch in range(self.batch_size):
+             # Get a new sample point
+             x_new = self._find_best_point(x_train, y_train_virtual)
 
-          # Evaluate the new sample point
-          y_new = self.prob.evaluate(np.atleast_2d(x_new))
+             # Get a virtual point
+             y_virtual = self._get_virtual_point(np.atleast_2d(x_new))
 
-          # Update training set
-          x_train = np.vstack([x_train, x_new])
-          y_train = np.vstack([y_train, y_new])
+             # Update training set with the virtual point
+             x_train         = np.vstack([x_train,         x_new    ])
+             y_train_virtual = np.vstack([y_train_virtual, y_virtual])
```

Comment:
if `batch_size == 1`, we don't need to compute the virtual point, do we?
No matter which `batch_type` is used, the virtual point only works with batch_size>=2, right?

```
x_new = self._find_best_point(x_train, y_train)
if self.batch_size > 1:
    y_train_virtual = y_train.copy()
    for batch in range(self.batch_size):
         # Get a virtual point
         y_virtual = self._get_virtual_point(np.atleast_2d(x_new))
         # Update training set with the virtual point
         x_train         = np.vstack([x_train,         x_new    ])
         y_train_virtual = np.vstack([y_train_virtual, y_virtual])
             
         x_new = self._find_best_point(x_train, y_train_virtual)
```

Reply:
This has now been address. The code was fine as is but it is not strictly necessary to call `_get_virtual_point` at the final point in the batch.